includes/class-wc-payments-token-service.php

```diff
@@ -100,7 +100,9 @@ public function add_payment_method_to_user( $payment_method_id, $user ) {
 	 * @return array
 	 */
 	public function woocommerce_get_customer_payment_tokens( $tokens, $user_id, $gateway_id ) {
-		if ( ( ! empty( $gateway_id ) && WC_Payment_Gateway_WCPay::GATEWAY_ID !== $gateway_id ) || ! is_user_logged_in() ) {
+		$gateway_ids = [ WC_Payment_Gateway_WCPay::GATEWAY_ID, WC_Payment_Gateway_WCPay::GATEWAY_ID . '_' . Payment_Method::SEPA ];
```

Comment:
Is there a utility function somewhere in WCPay to get the available gateway IDs so this doesn't have to be hard-coded? If so, we could delete this line and do the check in line 105 (this is pseudo-code):

`! in_array( $gateway_id, get_upe_gateway_ids, true )`

Reply:
This would require a function that returns all reusable gateway IDs. Such a function doesn't currently exist unfortunately. It might be good to have at some point though. For now, this the only place it would be used.